### PR TITLE
Add command line argument for custom TLS Passthrough port

### PR DESCRIPTION
**Pull request by haywoodsh**
### Proposed changes
Add flag for custom TLS Passthrough port.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: tests/data/common/service/loadbalancer-with-additional-ports.yaml [hunk 1]
```diff
@@ -19,6 +19,10 @@ spec:
     targetPort: 8080
     protocol: TCP
     name: api
+  - port: 8443
+    targetPort: 8443
+    protocol: TCP
+    name: custom-tsl-port
```
Comment (jjngx) [inline tests/data/common/service/loadbalancer-with-additional-ports.yaml:25]: typo?

File: docs/content/configuration/global-configuration/command-line-arguments.md [hunk 2]
```diff
@@ -94,6 +94,15 @@ See [-report-ingress-status](#cmdoption-report-ingress-status) flag.
...
 Enable TLS Passthrough on port 443.
 
+Requires [-enable-custom-resources](#cmdoption-enable-custom-resources).
+&nbsp;
+<a name="cmdoption-tls-passthrough-port"></a>
+
+### -tls-passthrough-port `<int>`
+
+Set the port for TLS Passthrough.
+Format: `[1 - 65535]` (default `443`)
```
Comment (shaun-nx) [inline docs/content/configuration/global-configuration/command-line-arguments.md:104]: Should this be `[1024 - 65535]` instead of `[1 - 65535]` ?

**Review by jjngx — APPROVED** [review]

File: tests/data/common/service/loadbalancer-with-additional-ports.yaml (hunk 1 above)
Comment (haywoodsh) [reply tests/data/common/service/loadbalancer-with-additional-ports.yaml]: fixed in the latest commit

File: cmd/nginx-ingress/flags.go [hunk 3]
```diff
@@ -164,7 +164,9 @@ var (
 		`The namespace/name of the GlobalConfiguration resource for global configuration of the Ingress Controller. Requires -enable-custom-resources. Format: <namespace>/<name>`)
 
 	enableTLSPassthrough = flag.Bool("enable-tls-passthrough", false,
-		"Enable TLS Passthrough on port 443. Requires -enable-custom-resources")
+		"Enable TLS Passthrough on default port 443. Requires -enable-custom-resources")
+
+	tlsPassthroughPort = flag.Int("tls-passthrough-port", 443, "Set the port for TLS Passthrough. [1024 - 65535]")
```
Comment (vepatel) [inline cmd/nginx-ingress/flags.go:169]:
```suggestion
	tlsPassthroughPort = flag.Int("tls-passthrough-port", 443, "Set custom port for TLS Passthrough. [1024 - 65535]")
```

File: internal/k8s/controller.go [hunk 4]
```diff
@@ -199,6 +199,7 @@ type NewLoadBalancerControllerInput struct {
 	IsPrometheusEnabled          bool
 	IsLatencyMetricsEnabled      bool
 	IsTLSPassthroughEnabled      bool
+	TLSPassthroughPort           int
```
Comment (vepatel) [inline internal/k8s/controller.go:202]: @haywoodsh  is there a way to assert config when `IsTLSPassthroughEnabled=false` and `TLSPassthroughPort` is set?

File: tests/suite/test_ts_tls_passthrough.py [hunk 5]
```diff
@@ -80,6 +85,7 @@ def fin():
 
 
 @pytest.mark.ts
+@pytest.mark.tls_passthrough
```
Comment (vepatel) [inline tests/suite/test_ts_tls_passthrough.py:88]: can we remove this marker?

Comment (haywoodsh) [reply tests/suite/test_ts_tls_passthrough.py:88]: Will remove this in the next commit

**Review by vepatel — APPROVED** [review]

**Review by ADubhlaoich — COMMENTED** [review]
LGTM where docs are concerned: there's some issues with product naming but they existed in the docs prior to change and will be reviewed and updated as a discrete documentation task.

**Review by shaun-nx — APPROVED** [review]
Really great stuff Haywood. The updated template tests are very clear and concise 👍

**Comment by jjngx** [comment]
👍🏻 great job @haywoodsh

**Pull request merged**